### ISSUE TITLE
False positive in mocha/handle-done-callback when done is called inside an object property callback

### ISSUE THREAD

**Issue by e22m4u** [lo1tuma/eslint-plugin-mocha#560]
After upgrading to eslint-plugin-mocha v12.0.1, the `mocha/handle-done-callback` rule reports an error when the done callback is invoked inside an arrow function that is passed as a property of an object. This worked fine in 11.0.3 versions.

```js
it('should detect done() inside an object property', function (done) {
  // A simple runner function
  function runTask(config) {
    config.onSuccess();
  }

  // Passing an object literal with a callback property
  runTask({
    onSuccess: () => {
      done(); // <--- Rule fails to detect this invocation in v12.0.1
      // error  Expected "done" callback to be handled  mocha/handle-done-callback
    },
  });
});
```

Real-world test-case:

```js
it('should provide the request context to the route handler', function (done) {
  const router = new TrieRouter();
  router.defineRoute({
    method: HttpMethod.GET,
    path: '/test',
    handler: ctx => {
      expect(ctx).to.be.instanceof(RequestContext);
      done();
    },
  });
  const req = createRequestMock({path: '/test'});
  const res = createResponseMock();
  router.handleRequest(req, res);
});
```
